### ISSUE TITLE
check and fix [OII] fluxes in latest ELG templates

### ISSUE THREAD

**Issue by moustakas** [desihub/desisim#47]
[From @sbailey ]

Christophe, Julien, and I are working on setting up some test data for a redshift fitting data challenge.  It seems that the [OII] flux in the new templates is too bright — when I integrate the [OII] doublet I get about a factor of 2.7x brighter than what is reported in the metadata table.

The distribution in the metadata table looks reasonable for DESI, while the templates themselves have whopping [OII] doublets that make it really really easy for the redshift fitters.  I suspect that the universe is not that kind.

```
import numpy as np
from desisim.templates import ELG
wave = np.arange(5000, 9800.1, 0.2)
flux, _, meta = ELG(nmodel=10, wave=wave).make_templates()
for i in range(len(meta)):
    z = meta['REDSHIFT'][i]
    ii = ((3727-2)*(1+z) < wave) & (wave < (3730+2)*(1+z))
    OIIflux = np.sum(flux[i,ii]*np.gradient(wave[ii]))
    print i, z, meta['OIIFLUX'][i], OIIflux
```

**Comment by moustakas** [comment]
I've confirmed that the EMSpectrum() class works as advertised (after a minor bug fix having to do with the default wavelength array which didn't affect other parts of the templates code):
```
import numpy as np
from desisim.templates import EMSpectrum
flux, wave, line = em.spectrum(oiiflux=1E-18)
ii = (wave > 3720) & (wave < 3740)
np.sum(flux[ii]*np.gradient(wave[ii]))
   1.0241218980875181e-18
```
Looking into the ELG() class now.  @sbailey I suspect that the "issue" is due to the fact that you're integrating the underlying stellar continuum as well as the [OII] line.  But let me confirm.

**Comment by moustakas** [comment]
From the desisim-templates branch (pending PR #49), this bit of code demonstrates that the emission-line strengths in the spectra now correctly match the line-strengths in the metadata table:
```
import numpy as np
from desisim.templates import ELG
wave = np.arange(5000, 9800.1, 0.2)
flux, _, meta = ELG(wave=wave).make_templates(nmodel=20,nocolorcuts=True,nocontinuum=True)
for i in range(len(meta)):
    z = meta['REDSHIFT'][i]
    ii = (3722*(1+z) < wave) & (wave < 3736*(1+z))
    OIIflux = np.sum(flux[i,ii]*np.gradient(wave[ii]))
    print(i, z, meta['OIIFLUX'][i], OIIflux, OIIflux/meta['OIIFLUX'][i])
(0, 0.67904013, 1.6908607e-16, 1.7155390246542051e-16, 1.0145951444796915)
(1, 0.9617359, 4.5936423e-17, 4.6492112413005104e-17, 1.0120969207128985)
(2, 1.1663995, 2.5609962e-17, 2.615764338494292e-17, 1.0213854920743415)
(3, 1.2280171, 1.0409711e-16, 1.0693576559149395e-16, 1.027269335457639)
(4, 1.0497733, 7.0756255e-17, 7.0109454692410111e-17, 0.99085874936340901)
(5, 0.98588037, 2.1249845e-16, 2.141664424149255e-16, 1.007849428746902)
(6, 1.5006541, 1.9333823e-16, 1.9463996278211466e-16, 1.0067329049382083)
(7, 1.4044151, 9.7202973e-17, 9.7922134489167334e-17, 1.0073985527686502)
(8, 0.95149022, 2.6117536e-16, 2.6164807747326725e-16, 1.0018099751570348)
(9, 1.0435394, 8.8888485e-17, 8.9102149628956269e-17, 1.0024037434354043)
(10, 1.5890381, 6.2851089e-17, 6.3123423828025841e-17, 1.0043330205961871)
(11, 1.226334, 2.2157065e-17, 2.1182942208684471e-17, 0.9560355661413722)
(12, 1.4615552, 5.9637274e-17, 5.9904079409361538e-17, 1.0044738077766053)
(13, 1.1088933, 3.7732885e-16, 3.7863855856732017e-16, 1.0034709942934104)
(14, 0.97485983, 2.6773362e-16, 2.6849932592289769e-16, 1.0028599728999188)
(15, 0.8395443, 5.217174e-17, 5.2299148213185232e-17, 1.0024420851582956)
(16, 1.0586351, 1.2900739e-16, 1.2919995311766391e-16, 1.00149263059538)
(17, 1.0496423, 6.0756112e-17, 6.0850566830679171e-17, 1.0015546623813389)
(18, 0.87181675, 1.8984752e-16, 1.9007903526141194e-16, 1.0012194831720223)
(19, 1.498772, 3.6603821e-17, 3.6612539761006211e-17, 1.0002381995058864)
```
Note that nocontinuum=True is critical for this test, otherwise you're also integrating the underlying stellar continuum, and you won't match meta['OIIFLUX'].  (The nocolorcuts=True is just a convenience to speed up the generation of the models.)

Also note that the only reason the flux ratio isn't *identically* unity for all the models is because of the variation in emission-line width (as well as the fact that there are other nearby Balmer lines).  For example, fixing the emission-line width ensures that the (fixed) integration window captures all of the flux:
```
import numpy as np
from desisim.templates import ELG
wave = np.arange(5000, 9800.1, 0.2)
flux, ww, meta = ELG(wave=wave).make_templates(nmodel=20,nocolorcuts=True,nocontinuum=True,linesigma_meansig=[np.log10(75),0.0])
for i in range(len(meta)):
    z = meta['REDSHIFT'][i]
    ii = (3722*(1+z) < wave) & (wave < 3736*(1+z))
    OIIflux = np.sum(flux[i,ii]*np.gradient(wave[ii]))
    print(i, z, meta['OIIFLUX'][i], OIIflux, OIIflux/meta['OIIFLUX'][i])
(0, 0.86783689, 1.6255715e-17, 1.6425788317238438e-17, 1.0104623491503124)
(1, 1.4201574, 6.7222441e-17, 6.8236292288855539e-17, 1.0150820434784416)
(2, 0.98512596, 1.8115441e-16, 1.828868495738263e-16, 1.0095633474697581)
(3, 1.5181663, 2.375383e-17, 2.395725195135398e-17, 1.0085637678853681)
(4, 0.63691258, 1.2504338e-16, 1.2575589280077188e-16, 1.0056980893620266)
(5, 0.74230635, 3.3666906e-17, 3.3925400794137522e-17, 1.0076780068753328)
(6, 1.5314953, 1.502702e-17, 1.5140117481160043e-17, 1.0075263008067343)
(7, 1.587365, 3.7200986e-17, 3.7544030284301259e-17, 1.00922138713725)
(8, 1.1760409, 2.0626644e-17, 2.0788726387808422e-17, 1.0078579340220957)
(9, 0.61583894, 8.4945963e-17, 8.5352985622839706e-17, 1.004791547012531)
(10, 1.0800118, 1.8746532e-17, 1.8856890758838559e-17, 1.0058869143001659)
(11, 0.76812845, 4.7461466e-17, 4.7763118791679716e-17, 1.0063557355932506)
(12, 0.86937386, 6.5125326e-17, 6.5534235360623711e-17, 1.0062788085549497)
(13, 1.1026907, 4.2866951e-17, 4.3060894428149026e-17, 1.0045243087579683)
(14, 1.1601127, 3.5590822e-17, 3.569688391450432e-17, 1.0029800275482759)
(15, 0.79602391, 4.9231768e-17, 4.9335263576276454e-17, 1.0021022024972559)
(16, 1.0683161, 2.2316631e-17, 2.2338296103626636e-17, 1.000970783880498)
(17, 1.2063377, 6.7218748e-17, 6.725351330179119e-17, 1.0005171967472724)
(18, 1.1571349, 3.3557447e-17, 3.3571793142148906e-17, 1.0004275128657854)
(19, 1.0634835, 3.4425081e-16, 3.4435202434102297e-16, 1.0002940052761717)
```

**Comment by sbailey** [comment]
Fixed in PR #49, including tests to verify that the integral of the spectrum over the [OII] doublet matches the meta table OIIFLUX to 1%, using the second example from @moustakas above.
